src/averageAges.js

```diff
@@ -34,7 +45,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let femaleArr = people.filter(x => x.sex === 'f');
+
+  if (withChildren) {
+    femaleArr = femaleArr.filter((x, i, arr) => {
+      return people.some(ch => ch.mother === x.name);
+    });
+  }
+
+  const averageAge = femaleArr.reduce((sum, item) =>
+    sum + (item.died - item.born), 0) / femaleArr.length;
```

Comment:
```suggestion
  const averageAge = femaleArr.reduce((sum, item) => (
    sum + (item.died - item.born)), 0
  ) / femaleArr.length;
```
your code should be readable